src/main/java/com/apinba/restapi/controllers/TeamController.java

```diff
@@ -2,47 +2,51 @@
 
 import com.apinba.restapi.models.TeamModel;
 import com.apinba.restapi.services.TeamService;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.*;
-
-import java.util.ArrayList;
-import java.util.Optional;
+import java.util.List;
 import java.util.UUID;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/teams")
 public class TeamController {
-    @Autowired
-    private TeamService teamService;
-
-    @GetMapping
-    public ArrayList<TeamModel> getTeams() {
-        return this.teamService.getTeams();
-    }
-
-    @PostMapping
-    public TeamModel saveTeam(@RequestBody TeamModel team) {
-        return this.teamService.saveTeam(team);
-    }
-
-    @GetMapping(path = "/{id}")
-    public Optional<TeamModel> getTeamById(@PathVariable UUID id){
-        return this.teamService.getById(id);
-    }
-
-    @PutMapping(path = "/{id}")
-    public TeamModel updateTeamById(@RequestBody TeamModel request, @PathVariable UUID id){
-        return this.teamService.updateById(request, id);
-    }
-
-    @DeleteMapping(path = "/{id}")
-    public String deleteTeamById(@PathVariable UUID id){
-        boolean ok = this.teamService.deleteTeamById(id);
 
-        if(ok){
-            return "Team with id " + id + "DELETED!";
-        } else {
-            return "Team with id " + id + "HAVEN'T DELETED PA NO SE DELETIO!";
-        }
+  private final TeamService teamService;
+
+  public TeamController(TeamService teamService) {
+    this.teamService = teamService;
+  }
+
+  @GetMapping
+  public List<TeamModel> getTeams() {
+    return this.teamService.getTeams();
+  }
+
+  @PostMapping
+  public TeamModel saveTeam(@RequestBody TeamModel team) {
+    return this.teamService.saveTeam(team);
+  }
+
+  @GetMapping(path = "/{id}")
+  public ResponseEntity<TeamModel> getTeamById(@PathVariable UUID id) {
+    return ResponseEntity.of(teamService.getById(id));
```

Comment:
El método `ResponseEntity.of()` recibe un `Optional`. Si tiene contenido, nuestra controller retorna `200 OK`, si esta vacío, `404 Not Found`